### PR TITLE
audit.rb: Check for signature with sptcl

### PR DESCRIPTION
**Pull request by vitorgalvao**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Seems like signature verification [is occasionally failing](https://github.com/Homebrew/homebrew-cask/pull/145059#issuecomment-1507351629).

Over many trials I’ve settled on this `spctl` method as the most reliable method to check if an app or binary is signed and notarised. Haven’t tried extensively on PKGs but it should work the same. Submitting the PR seemed the most efficient way to make the suggestion and allow others to try it.

Add `-vv` to see details. Example output:

```
/Volumes/Garmin Express/Install Garmin Express.pkg: accepted
source=Notarized Developer ID
origin=Developer ID Installer: Garmin International (72ES32VZUA)
```

### CODE REVIEW

**Review by MikeMcQuaid — APPROVED** [review]
This makes sense to me, interested in opinions from @reitermarkus @SMillerDev too.

**Comment by reitermarkus** [comment]
@Homebrew/cask, what was the specific error here? Does `codesign` or `pkgutil` fail?

**Comment by reitermarkus** [comment]
This is the error:

```
Install Synology Drive Client.pkg does not have a ticket stapled to it.
```

**Comment by reitermarkus** [comment]
So based on the message I assume this comes from the `stapler` command, which isn't changed in this PR.

**Comment by vitorgalvao** [comment]
Not having a stapled ticket shouldn’t be a cause for failure. Stapling just saves the need to do the initial download of the ticket, which takes less than a second, happens on first run, is irrelevant for PKGs (the slight delay doesn’t matter for an installer), and not available on raw binaries. This command takes the notarisation into account regardless of stapling.

I’ve needed this kind of verification extensively outside of the Homebrew context and this command has so far emerged as the clear winner. Your needs for the audit may be different, so merge, edit, or close at will.

**Comment by reitermarkus** [comment]
> Not having a stapled ticket shouldn’t be a cause for failure.

Okay, so should we remove the `stapler` check any only use `spctl` which seems to work for apps and pkgs?

**Comment by Bo98** [comment]
stapler requires an Xcode installation so removing any use that would be great to address #14949.

File: Library/Homebrew/cask/audit.rb [hunk 1]
```diff
@@ -474,55 +474,18 @@ def audit_signing
...
-            next if result.success?
-
-            add_error(<<~EOS, strict_only: true)
-              Signature verification failed:
-              #{result.merged_output}
-              macOS on ARM requires applications to be signed.
-              Please contact the upstream developer to let them know they should notarize their package.
-            EOS
-          end
+          path = tmpdir/artifact.source.relative_path_from(cask.staged_path)
```
[the remark this answers is not in the document]
Comment (vitorgalvao) [reply Library/Homebrew/cask/audit.rb]: Fixed. A multiline ternary looks better, but `style` doesn’t like that:

```ruby
path = artifact.is_a(Artifact::Pkg) ?
  tmpdir/artifact.path.relative_path_from(cask.staged_path) :
  tmpdir/artifact.source.relative_path_from(cask.staged_path)
```

[the remark this answers is not in the document]
Comment (vitorgalvao) [reply Library/Homebrew/cask/audit.rb]: Sure. Done.

**Review by reitermarkus — APPROVED** [review]

**Comment by reitermarkus** [comment]
Thanks!

**Pull request merged**